x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/RuntimeKeywordMappedFieldType.java

```diff
@@ -77,8 +78,17 @@ public String typeName() {
         return scriptFactory.newFactory(script.getParams(), context.lookup());
     }
 
+    private void checkAllowExpensiveQueries(QueryShardContext context) {
+        if (context.allowExpensiveQueries() == false) {
+            throw new IllegalArgumentException(
+                "queries cannot be executed against [script] fields while [" + ALLOW_EXPENSIVE_QUERIES.getKey() + "] is set to [false]."
+            );
+        }
+    }
+
     @Override
     public Query existsQuery(QueryShardContext context) {
+        checkAllowExpensiveQueries(context);
```

Comment:
maybe one day we will a base class for runtime mapper field types that does this in one place.

Reply:
probably!